src/coreclr/vm/ceeload.h

```diff
@@ -339,6 +339,10 @@ struct VASigCookie
     Volatile<PCODE> pNDirectILStub;         // will be use if target is NDirect (tag == 0)
     PTR_Module      pModule;
     Signature       signature;
+    TypeHandle*     classInst;
```

Comment:
This needs to handle unloadability. The module may be non-collectible, but one of the instantiation arguments may be collectible. The whole thing needs to be collectible in this case.

Reply:
Added the logic to handle unloadability. Not sure if it's the best way though.